.github/workflows/auto-approve.yml

```diff
@@ -1,13 +1,75 @@
-name: Auto approve
-
+name: auto approve and merge and tidy
 on:
-  pull_request
+  pull_request:
+    paths:
+      - ".github/workflows/gosum.yml"
+      - "go.mod"
+      - "go.sum"
 
+# job level conditions don't seem to work at the moment
+# https://github.community/t5/GitHub-Actions/Status-of-workflows-with-no-running-jobs/td-p/37160
+# which is why github.actor == 'dependabot[bot]' || github.actor == 'dependabot-preview[bot]' is
+# repeated over and over
 jobs:
-  auto-approve:
+  tidy:
+    name: run go mod tidy and updated
+    runs-on: ubuntu-latest
+    steps:
+      - name: checkout
+        uses: actions/checkout@v1
+        if: github.actor == 'dependabot[bot]' || github.actor == 'dependabot-preview[bot]'
+      - name: reattach HEAD to Head Ref
+        # b/c checkout action leaves in detached head state https://github.com/actions/checkout/issues/6
+        run: git checkout "$(echo ${{ github.head_ref }})"
+        if: github.head_ref != '' && (github.actor == 'dependabot[bot]' || github.actor == 'dependabot-preview[bot]')
+      - name: setup go
+        uses: actions/setup-go@v1
+        with:
+          go-version: "1.14"
+      - name: Tidy
+        run: |
+          go version
+          go mod tidy
+        if: github.actor == 'dependabot[bot]' || github.actor == 'dependabot-preview[bot]'
+      - name: set up Git
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        run: |
+          git config user.name "${GITHUB_ACTOR}"
+          git config user.email "${GITHUB_ACTOR}@users.noreply.github.com"
+          git remote set-url origin https://x-access-token:${GITHUB_TOKEN}@github.com/${GITHUB_REPOSITORY}.git
+        if: github.actor == 'dependabot[bot]' || github.actor == 'dependabot-preview[bot]'
```

Comment:
Can you put this `if` statement at the top? I feel like I remember Github fixing it so you could. Meaning you can have:

```
  tidy:
    name: run go mod tidy and updated
    runs-on: ubuntu-latest
    if: github.actor == 'dependabot[bot]' || github.actor == 'dependabot-preview[bot]'
...
```

If that works you can reduce a ton copied code.

Reply:
ooh will try